src/Samples/PsdTgaExporter.cpp

```diff
@@ -80,12 +87,22 @@ namespace tgaExporter
 #if PSD_USE_MSVC
 		const errno_t success = _wfopen_s(&file, filename, L"wb");
         if (success != 0)
-#else
+#elif defined(__APPLE__)
         std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>,wchar_t> convert;
         std::string s = convert.to_bytes(filename);
         char const *cs = s.c_str();
         file = fopen(cs, "wb");
         if (file == NULL)
+#else
+		//In Linux
+		char *buffer;
+		size_t n = std::wcslen(filename) * 4 + 1;
+		buffer = static_cast<char*>(std::malloc(n));
+		std::memset(buffer,0,n);
+		std::wcstombs(buffer,filename,n);
+		file = fopen(buffer,"wb");
+		std::free(buffer);
+		if (file == NULL)
 #endif

```

Comment:
Please use #elif defined(__linux) (or appropriate) in this case, so we can output an #error in the #else case.